membership_file/tests/test_middleware.py

```diff
@@ -0,0 +1,43 @@
+from django.contrib.auth.models import User, AnonymousUser
+from django.test import TestCase, RequestFactory
+
+from membership_file.exceptions import UserIsNotCurrentMember
+from membership_file.middleware import MembershipMiddleware
+from membership_file.models import Member
+
+
+class MembershipMiddlewareTestCase(TestCase):
+    fixtures = ['test_users', 'test_members']
+
+    def setUp(self):
+        self.middleware = MembershipMiddleware()
+
+    def _create_request(self, user_id):
+        request = RequestFactory().get('')
+        request.user = User.objects.get(id=user_id)
+        return request
+
+    def test_process_request_not_a_member(self):
+        request = self._create_request(2)
+        self.middleware.process_request(request)
+        self.assertEqual(request.member, None)
+
+    def test_process_request_is_a_member(self):
+        request = self._create_request(100)
+        self.middleware.process_request(request)
+        self.assertEqual(request.member, Member.objects.get(user__id=100))
+
+    def test_process_request_anonymoususer(self):
+        request = RequestFactory().get('')
+        request.user = AnonymousUser()
+        self.middleware.process_request(request)
+        self.assertEqual(request.member, None)
+
+    def test_process_exception(self):
+        request = self._create_request(1)
+        self.assertIsNone(self.middleware.process_exception(request, Exception()))  # Ignore irrelevant errors
```

Comment:
Which “irrelevant errors” are you ignoring?  
Or are you checking if there are no other errors?

Reply:
Here I want to intercept only UserIsNotAMember exceptions, any other exception should be ignored (eg HTTP404 exception). I want specific behaviour in case that exception is thrown (return that template). Check the Django Middleware documentation on what Middleware does if you want to know why and how this method is triggered https://docs.djangoproject.com/en/dev/topics/http/middleware/#process-exception

Reply:
> any other exception should be ignored

With ignored, you mean catch and ignore or just letting it be thrown?

Reply:
Okay, would this be a correct comment?

```suggestion
        self.assertIsNone(self.middleware.process_exception(request, Exception()))  # Check if irrelevant errors are ignored
```